packages/flutter/lib/src/widgets/shared_state.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+
+/// An inherited widget to manage shared state for its descendants.
+///
+/// This can be used for managing state when multiple widgets need their state
+/// to be kept in sync. It is most useful when the closest common widget to the
+/// widgets that need to share the state is several levels above the widgets
+/// that are using it. When the state is close to the widgets using them, it is
+/// often preferable to pass the needed state to the descendants that need it
+/// directly instead of using this widget.
+///
+/// For complex applications, it may be more appropriate to use a package that
+/// provides more features and a stream-oriented paradigm like Rx.
+///
+/// This widget doesn't keep the state itself, it uses the state passed in. It
+/// calls the [valueChanged] callback when the state changes. You should then
+/// pass the updated state to this widget on the next build as the [value].
+///
+/// When the shared state's [value] is updated, all of the widgets that depend
+/// upon it will be rebuilt.
+///
+/// ## Sample code
+///
+/// ```dart
+/// enum Thing {
+///   raindrops,
+///   whiskers,
+///   kettles,
+///   mittens,
+///   kittens,
+///   packages,
+///   string,
+/// }
+/// 
+/// class FavoritePage extends StatefulWidget {
+///   const FavoritePage({Key key}) : super(key: key);
+/// 
+///   @override
+///   _FavoritePageState createState() => new _FavoritePageState();
+/// }
+/// 
+/// class _FavoritePageState extends State<FavoritePage> {
+///   Thing _favoriteThing;
+/// 
+///   @override
+///   Widget build(BuildContext context) {
+///     return new SharedState<Thing>(
+///       value: _favoriteThing,
+///       valueChanged: (Thing thing) {
+///         if (_favoriteThing != thing) {
+///           setState(() {
```

Comment:
It's odd that FavoritePage would rebuild if Thing changed since it doesn't use that value.  I don't think this is a good example

Reply:
I think the point of using an inherited widget is that FavoritePage doesn't care about _favoriteThing changing, its children do.  So we shouldn't be called setState